src/coreclr/jit/helperexpansion.cpp

```diff
@@ -795,22 +795,26 @@ PhaseStatus Compiler::fgExpandStaticInit()
         return result;
     }
 
-    if (opts.OptimizationDisabled())
+    // Always expand for NativeAOT, see
+    // https://github.com/dotnet/runtime/issues/68278#issuecomment-1543322819
+    const bool isNativeAOT = IsTargetAbi(CORINFO_NATIVEAOT_ABI);
+
+    if (!isNativeAOT && opts.OptimizationDisabled())
```

Comment:
I would just have both the conditions under single if

```c++
if (!isNativeAOT)
{
  if (opts.OptimizationDisabled()) {}
  if(preferSize) {}
}
```

Reply:
removed "preferSize" since it was introduced for NAOT specifically and no longer needed